### ISSUE TITLE
When .fit(), Ti is shifted by 1 month

### ISSUE THREAD

**Issue by juanbretti** [scuervo91/dcapy#21]
Hello,
Thank you very much for this package. Is great.

I just have one issue.
For some reason, the `Ti` is shifted by 1 month when doing `.fit()`.

Here is a simple example:

```python
import pandas as pd
import numpy as np
import dcapy.dca as dca  # https://scuervo91.github.io/dcapy/
import datetime

df = pd.DataFrame({
    'date': ['2020-01-01', '2020-02-01', '2020-03-01', '2020-04-01', '2020-05-01'],
    'prod': [10, 9, 8, 7, 6]})
df['date'] = pd.to_datetime(df['date'], format='%Y-%m-%d')

# df.plot(x='date',y='prod', kind='scatter')

d1 = dca.Arps(freq_di='M')
d1.fit(df=df, time='date', rate='prod')
d1
```

The output I get is:

```
Declination 
 Ti: 2020-02-01 
 Qi: 9.044496325102125 bbl/d 
 Di: 0.004430541919967859 FreqEnum.M 
 b: 4.647764695265519e-08
```

As you can see, the `Ti` is 1/FEB/2020, but at the dataset, the first month was 1/JAN/2020.
Also, notice `Qi` is ~9 instead of the original 10. Which is the oil_rate for the predicted 1/FEB/2020.

Is there any reason why?
Thanks!

Here is another short example:

```python
a3 = dca.Arps(
    ti = datetime.date(2021,1,1),
    di = 0.03,
    qi = 10,
    b = 0,
    freq_di='M',
    seed=42
)

a4 = a3.forecast(start=datetime.date(2021,1,1),end=datetime.date(2022,1,1), freq_output='M', iter=10)
a4 = a4.reset_index()
a4['date'] = a4['date'].astype('datetime64[ns]') 

d2 = dca.Arps(freq_di='M')
d2.fit(df=a4, time='date', rate='oil_rate', b=0)
d2
```

The output
```
Declination 
 Ti: 2021-02-01 
 Qi: 9.694755946748566 bbl/d 
 Di: 1.0 FreqEnum.M 
 b: 0.0
```

**Comment by juanbretti** [comment]
I found the issue is at `arps.py` at the `fit()` method and the interaction with the filters.
To quick and dirty fix the issue, I am adding an additional element to the array for `x` and `yb`.

Previously was:
```python
        # TODO: Add the option to start the cumulative with an Initial Value different a 0
        #Check inputs
        x = df[time].values if isinstance(time,str) else time 
        yb = df[rate].values if isinstance(rate,str) else rate
```

Now is:
```python
        # TODO: Add the option to start the cumulative with an Initial Value different a 0
        #Check inputs
        x = df[time].values if isinstance(time,str) else time 
        yb = df[rate].values if isinstance(rate,str) else rate
        
        # TODO: JPBM: Fix?
        if isinstance(x[0],(np.datetime64,date)):
            x_delta = np.timedelta64(1, 'M').astype('<m8[ns]')
        else:
            x_delta = 1.
        x_previous = np.min(x) - x_delta
        x = np.append(x_previous, x)
        yb = np.append(0, yb)
```

I need to spend more time to understand how the filtering works.

**Comment by juanbretti** [comment]
By the way, with this toy example:

```python
import pandas as pd
import numpy as np
import dcapy.dca as dca  # https://scuervo91.github.io/dcapy/
import datetime
import matplotlib.pyplot as plt

a3 = dca.Arps(
    ti = datetime.date(2021,1,1),
    qi = 20,
    di = 0.1,
    b = 0.1,
    freq_di='M',
    seed=42
)

a4 = a3.forecast(start=datetime.date(2021,1,1),end=datetime.date(2022,1,1), freq_output='M', iter=10)
a4 = a4.reset_index()
a4['date'] = a4['date'].astype('datetime64[ns]') 

d2 = dca.Arps(freq_di='M')
d2.fit(df=a4, time='date', rate='oil_rate')
print(d2)

fig, ax = plt.subplots(figsize=(10,7))
a4.plot(x='date', y='oil_rate', kind='scatter',ax=ax)
d2.plot(ax=ax, start=datetime.date(2021,1,1), end=datetime.date(2022,1,1))
```

I get this output:
```
Declination 
 Ti: 2021-01-01 
 Qi: 20.000000000000004 bbl/d 
 Di: 0.0033333333333333375 M 
 b: 0.10000000000000345
```

![image](https://github.com/scuervo91/dcapy/assets/10165911/6e93bd1f-7f54-4aa2-9765-13001964714f)

Which still not a perfect match. The `Di` is not fitting correctly.
Not sure if this issue with the date has to do with it.

**Comment by scuervo91** [comment]
Fix with #23
